fix: make all attribute groups except for wps single column

b12f opened the pull request:
Ref: https://community.openproject.org/work_packages/49400/activity

Commit:
fix: make all attribute groups except for wps single column

Ref: https://community.openproject.org/work_packages/49400/activity

HDinger commented on frontend/src/global_styles/content/_attributes_key_value.sass:
```diff
@@ -31,11 +31,6 @@
-  flex-shrink: 1
-  min-width: calc(50% - 3rem)
-  max-width: calc(100% - 3rem)
   margin: 0 1.5rem
```
As the attributes are now all below each other we can get rid of this extra margin and the magical calculation done here: https://github.com/opf/openproject/blob/a705bc530f4a2bd87d0f8e03ae4244b9b8b82bca/frontend/src/global_styles/content/_attributes_group.sass#L85C1-L90C1

HDinger commented on frontend/src/app/features/work-packages/components/wp-form-group/wp-attribute-group.component.sass:
```diff
@@ -1,8 +1,47 @@
 @import "src/assets/sass/helpers"
 
 .wp-attribute-group
+  display: flex
```
Wouldn't it be better to use a `grid` here?  Thus we could avoud all these magic calculations

b12f replied:
Unfotunately no, because of the requirement that the the fields go single column at some point, but we cannot use breakpoints since the width of the layout is independent of the width of the screen. You can show/hide and drag the sidemenu and split screen.

HDinger reviewed (changes requested):
It works fine. 👍 I have some remarks to the code.

Commit:
refactor: move confusing key value spacing to wp-attribute-group

Single-column attribute groups don't need the special gap spacing, so that
CSS is moved into the wp-attribute-groups.

HDinger committed:
Use grid instead of flex to avoid magic width caculations

HDinger reviewed (approved):
@b12f  I changed the wp attribute to a grid. I _think_ that I catched all cases, but please double check that everything works as expected before merging.

b12f commented:
Hi @HDinger unfortunately this breaks mobile screens of less than 432px width because the 400px minimum are a hard limit:

![20230808_10h53m08s_grim](https://github.com/opf/openproject/assets/11312895/1102fd33-4d90-400b-9659-f163ff301104)

HDinger committed:
Add special rule for mobile behavior

Commit:
Revert "Add special rule for mobile behavior"

This reverts commit 56d2657c4bbad1e4e755b2636cc8d22817063360.

Commit:
Revert "Use grid instead of flex to avoid magic width caculations"

This reverts commit 21360e16f7f6026c5b23d22395c96169950c82c7.

Commit:
docs: add explanatory comments to wp-attribute-group

After trying to refactor the wp-attribute-group styling to use CSS grid
instead of flex, we've decided to return to flexbox, since not all
requirements could be captured by grids.

However, the original problem still stands; the CSS flex solution
uses quite a bit of calculations and magic and is thus hard to grasp.

Henriette Darge <h.darge@openproject.com>
